firedrake/slate/slac/kernel_builder.py

```diff
@@ -27,52 +25,53 @@ def __init__(self, expression, tsfc_parameters=None):
                               TSFC when constructing subkernels associated
                               with the expression.
         """
-        assert isinstance(expression, TensorBase)
+        assert isinstance(expression, slate.TensorBase)
 
         if expression.ufl_domain().variable_layers:
             raise NotImplementedError("Variable layers not yet handled in Slate.")
 
-        ref_counts = collect_reference_count([expression])
-
-        # Collect terminals and expressions
+        # Collect terminals, expressions, and reference counts
         temps = OrderedDict()
         action_coeffs = OrderedDict()
-        tensor_ops = []
         seen_coeff = set()
-        for tensor in traverse_dags([expression]):
-            if isinstance(tensor, Tensor):
+        expression_dag = expand_dag(expression)
```

Comment:
FWIW, the name `expand_dag` has negative connotations, you're not actually expanding the dag, you're just eagerly evaluating the traversal of the dag.

Reply:
I see. So I guess the old name is better? I guess I just misunderstood.